js/components/MilkdropWindow/index.js

```diff
@@ -103,6 +103,33 @@ class MilkdropWindow extends React.Component {
       }
     }
   }
+  _handleKeyboardInput(e) {
+    if (e.which === 32) {
+      this._nextPreset(USER_PRESET_TRANSITION_SECONDS);
+    } else if (e.which === 8) {
+      this._prevPreset(0);
+    } else if (e.which === 72) {
+      this._nextPreset(0);
+    }
+  }
+  _nextPreset(blendTime) {
+    const presetIdx = Math.floor(this.presetKeys.length * Math.random());
```

Comment:
Is the "next" action in Winamp actually random?

Reply:
by default, yes! hahah
Some more common keyboard commands we can quickly add after this if we want:
R: toggles between random and sequential
Scroll Lock: stops preset cycling

Reply:
oh yea, definitely, good catch